motionplan/check.go

```diff
@@ -195,17 +165,18 @@ func checkPlanRelative(
 			return err
 		}
 		thisArcEndPose := spatialmath.Compose(thisArcEndPoseInWorld.Pose(), errorState)
-		startInputs := plan.Trajectory()[i-1]
-		nextInputs := plan.Trajectory()[i]
-		segment, err := createSegment(sf, lastArcEndPose, thisArcEndPose, startInputs, nextInputs)
-		if err != nil {
-			return err
+		segment := &ik.Segment{
+			StartPosition:      lastArcEndPose,
+			EndPosition:        thisArcEndPose,
+			StartConfiguration: plan.Trajectory()[i-1][checkFrame.Name()],
```

Comment:
I think this can be done more safely. Is it possible that `plan.Trajectory()[i-1][checkFrame.Name()]` is nil since for some reason `checkFrame.Name()` does not exist? Perhaps we can do `path.Trajectory().GetFrameInputs(checkFrame.Name())`